MinGW CI fails to build libpng

See https://github.com/gbdev/rgbds/runs/2039544097?check_suite_focus=true#step:5:316

Not *that* big of a deal, since we should be able to rely on MSVC instead of the time being, but it still wrongly flags CI builds as failed.

8a75cc50518d3d7bad9f7110fa2d13bf3ccc74a4 (2021-03-03) succeeded: https://github.com/gbdev/rgbds/runs/2024246118

a4049a3f1b58bbc699b86808bb2fe6dcb692fefc (2021-03-05) failed: https://github.com/gbdev/rgbds/runs/2039544097

Both use `libpng-1.6.37.tar.xz` and `libpng-1.6.37-apng.patch.gz`, although the particular Sourceforge DL servers are different.

Some slight details change in the build log, like utils being in `/usr/bin` instead of `/bin`; was an update to Github's Actions environment responsible?

The particular error:

```
libtool: link:  i686-w64-mingw32-gcc -shared .libs/libpng16-16.dll.def  .libs/png.o .libs/pngerror.o .libs/pngget.o .libs/pngmem.o .libs/pngpread.o .libs/pngread.o .libs/pngrio.o .libs/pngrtran.o .libs/pngrutil.o .libs/pngset.o .libs/pngtrans.o .libs/pngwio.o .libs/pngwrite.o .libs/pngwtran.o .libs/pngwutil.o mips/.libs/mips_init.o mips/.libs/filter_msa_intrinsics.o intel/.libs/intel_init.o intel/.libs/filter_sse2_intrinsics.o powerpc/.libs/powerpc_init.o powerpc/.libs/filter_vsx_intrinsics.o   -lz  -O2   -o .libs/libpng16-16.dll -Wl,--enable-auto-image-base -Xlinker --out-implib -Xlinker .libs/libpng16.dll.a
/usr/bin/i686-w64-mingw32-ld: .libs/pngrutil.o:pngrutil.c:(.text+0x46f8): undefined reference to `__memcpy_chk'
/usr/bin/i686-w64-mingw32-ld: .libs/pngwutil.o:pngwutil.c:(.text+0x262f): undefined reference to `__memcpy_chk'
collect2: error: ld returned 1 exit status
```

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -133,7 +133,7 @@ jobs:
     strategy:
       matrix:
         bits: [32, 64]
-        os: [ubuntu-latest]
+        os: [ubuntu-18.04]
         include:
           - bits: 32
             arch: i686

